frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -31,10 +31,17 @@ export function isArgoWorkflowTemplate(template: Workflow): boolean {
 }
 
 // Assuming template is the JSON format of PipelineSpec in api/v2alpha1/pipeline_spec.proto
-export function convertJsonToV2PipelineSpec(template: string): PipelineSpec {
-  const pipelineSpecJSON = JSON.parse(template);
-  const ts_pipelinespec = PipelineSpec.fromJSON(pipelineSpecJSON);
-  return ts_pipelinespec;
+export function convertYamlToV2PipelineSpec(template: string): PipelineSpec {
+  // const pipelineSpecJSON = JSON.parse(template);
+  // const ts_pipelinespec = PipelineSpec.fromJSON(pipelineSpecJSON);
+  // return ts_pipelinespec;
+  const pipelineSpecYAML = jsyaml.safeLoad(template);
+  const ts_pipelinespec = PipelineSpec.fromJSON(pipelineSpecYAML);
+  if (!ts_pipelinespec.root || !ts_pipelinespec.pipelineInfo || !ts_pipelinespec.deploymentSpec) {
+    throw new Error('Important infomation is missing. Pipeline Spec is invalid.');
+  } else {
```

Comment:
You can remove `else` here. That means you can return `ts_pipelinespec` as long as important information is not missing.

Reply:
Oh, I didn't aware this. Thanks for the suggestions! @zijianjoy